pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1020,6 +1020,68 @@ func TestMaxUnavailable(t *testing.T) {
 	}
 }
 
+func TestGetNonNegativeIntFromAnnotation(t *testing.T) {
+	tests := []struct {
+		name          string
+		annotations   map[string]string
+		expectedValue int32
+		expectedValid bool
+		expectErr     bool
```

Comment:
nit: Any reason not to assert the error message? it might be helpful to see what is going to be logged in each case.